File: data-pipeline/telemetry-cleaner/src/main/java/org/ekstep/ep/samza/Event.java
package org.ekstep.ep.samza;

import java.util.Map;


public class Event {
    private final Map<String, Object> map;

    public Map<String, Object> getMap() {
        return map;
    }

    public Event(Map<String, Object> map) {
        this.map = map;
    }

    public String eid() {
        return map != null && map.containsKey("eid") ? (String) map.get("eid") : null;
    }
}

File: data-pipeline-flink/druid-events-validator/src/main/scala/org/sunbird/dp/validator/domain/Event.scala
package org.sunbird.dp.validator.domain

import java.util

import org.sunbird.dp.core.domain.Events

class Event(eventMap: util.Map[String, Any]) extends Events(eventMap) {

  private val jobName = "DruidValidator"
  
  def markValidationSuccess(): Unit = {
    this.updateFlags("dv_processed", true)
  }

  def markValidationFailure(errorMsg: String): Unit = {
    this.updateFlags("dv_processed", false)
    this.updateFlags("dv_validation_failed", true)
    telemetry.addFieldIfAbsent("metadata", new util.HashMap[String, AnyRef])
    if (null != errorMsg) {
      telemetry.add("metadata.dv_error", errorMsg)
      telemetry.add("metadata.src", jobName)
    }
  }

  def isSummaryEvent: Boolean = "ME_WORKFLOW_SUMMARY".equalsIgnoreCase(eid)

  def isSearchEvent: Boolean = "SEARCH".equalsIgnoreCase(eid)

}
